src/nni_evaluation_engine.cpp

```diff
@@ -132,16 +130,18 @@ void NNIEvalEngineViaGP::UpdateEngineAfterModifyingDAG(
   GetGPEngine().ProcessOperations(GetDAG().PopulatePLVs());
 
   // Optimize branch lengths.
-  if (IsOptimizeNewEdges()) {
-    for (const auto &[pre_nni, nni] : pre_nni_to_nni) {
-      std::ignore = pre_nni;
-      NNIBranchLengthOptimization(nni, new_edge_ids);
-    }
-    GetGPEngine().ProcessOperations(GetDAG().PopulatePLVs());
+  if (optimize_new_edges_) {
+    // for (const auto &[pre_nni, nni] : pre_nni_to_nni) {
+    //   std::ignore = pre_nni;
+    //   NNIBranchLengthOptimization(nni, new_edge_ids);
+    // }
+    // GetGPEngine().ProcessOperations(GetDAG().PopulatePLVs());
+    BranchLengthOptimization();
   }
 
   GetGPEngine().ProcessOperations(GetDAG().ComputeLikelihoods());
   auto likelihoods = GetGPEngine().GetPerGPCSPLogLikelihoods();
+  std::cout << "GPEngine::UpdateEngineAfterModifying [END]" << std::endl;
```

Comment:
Let's either remove the printed message or make it optional with an is_quiet parameter.